mcu/atmega-hal/src/adc.rs

```diff
@@ -1,4 +1,28 @@
 //! Analog-to-Digital Converter
+//!
+//! # Example
+//!
+//! Complete example source code can be found in the
+//! [repository](https://github.com/Rahix/avr-hal/blob/main/examples/atmega2560/src/bin/atmega2560-adc.rs).
```

Comment:
Also minor, but i'd prefer to have the link be the filename of the example so it's easier to reference offline:

```suggestion
//! Complete example source code can be found in the repository:
//! [`atmega2560-adc.rs`](https://github.com/Rahix/avr-hal/blob/main/examples/atmega2560/src/bin/atmega2560-adc.rs)
```

Same for all the other examples, of course.